components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/internal/impl/UserSessionManagementServiceImpl.java

```diff
@@ -317,6 +331,42 @@ public boolean terminateSessionsByUserId(String userId) throws SessionManagement
         return true;
     }
 
+    /**
+     * Filter out the session IDs that are only applicable for the given role.
+     *
+     * @param sessionIdList List of session IDs.
+     * @param roleId        Role ID.
+     * @return List of filtered session IDs.
+     */
+    private List<String> filterSessionIdListByRole(List<String> sessionIdList, String roleId)
+            throws SessionManagementServerException {
+
+        try {
+            RoleManagementService roleManagementServiceV2 = FrameworkServiceDataHolder.getInstance()
+                    .getRoleManagementServiceV2();
+            Role role = roleManagementServiceV2.getRole(roleId);
+            String associatedApplication = role.getAudienceName();
```

Comment:
This may need only for application audience roles. Hence let's check the audience type and return if it is organization audience.